File: backend/app/schemas/ai.py
from pydantic import BaseModel
from typing import Any

# ── Segment intent ────────────────────────────────────────────────────────────

class IntentRequest(BaseModel):
    prompt: str
    context: dict[str, Any] | None = None

class IntentParseResult(BaseModel):
    segment_rules: dict[str, Any]
    segment_name: str
    message_draft: str
    recommended_channel: str
    reasoning: str

# ── Message draft ─────────────────────────────────────────────────────────────

class MessageDraftRequest(BaseModel):
    channel: str
    audience_description: str
    brand_context: str | None = "Aura Beauty — premium Indian beauty brand"
    tone: str | None = None

# ── Chat ──────────────────────────────────────────────────────────────────────

class ChatMessage(BaseModel):
    role: str          # "user" | "assistant"
    content: str

class ChatRequest(BaseModel):
    messages: list[ChatMessage]

File: frontend/src/store/ai.ts
import { create } from 'zustand';

// ── Types ─────────────────────────────────────────────────────────────────────

export interface IntentResult {
  segment_rules: {
    operator: 'AND' | 'OR';
    rules: Array<{ field: string; op: string; value: string | number }>;
  };
  segment_name: string;
  message_draft: string;
  recommended_channel: string;
  reasoning: string;
}

export interface ChatMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  timestamp: Date;
  /** Populated when AI returns structured IntentParseResult */
  result?: IntentResult;
}

interface AIState {
  messages: ChatMessage[];
  isStreaming: boolean;

  // Actions
  addMessage: (msg: Omit<ChatMessage, 'id' | 'timestamp'>) => string;
  updateLastMessage: (content: string, result?: IntentResult) => void;
  clearHistory: () => void;
  setStreaming: (streaming: boolean) => void;
}

// ── Store ─────────────────────────────────────────────────────────────────────

/**
 * AI chat store — session-only (no persistence).
 * Chat history is cleared on page refresh, matching the user expectation for
 * a conversational, stateless AI assistant.
 */
export const useAIStore = create<AIState>()((set) => ({
  messages: [],
  isStreaming: false,

  addMessage: (msg) => {
    const id = crypto.randomUUID();
    const message: ChatMessage = {
      ...msg,
      id,
      timestamp: new Date(),
    };
    set((state) => ({ messages: [...state.messages, message] }));
    return id;
  },

  updateLastMessage: (content, result) => {
    set((state) => {
      const messages = [...state.messages];
      const lastIdx = messages.length - 1;
      if (lastIdx < 0) return state;
      messages[lastIdx] = { ...messages[lastIdx], content, result };
      return { messages };
    });
  },

  clearHistory: () => set({ messages: [], isStreaming: false }),

  setStreaming: (streaming) => set({ isStreaming: streaming }),
}));
